### ISSUE TITLE
The primary text in Granite.HeaderLabel misaligned when placed in Gtk.Popover

### ISSUE THREAD

**Issue by ryonakano** [elementary/stylesheet#1244]
### What Happened?

![2023-03-04 13 09 10 のスクリーンショット](https://user-images.githubusercontent.com/26003928/222875108-9c421462-46f1-4afd-98d3-516e1e551c37.png)

The primary and secondary text in Granite.HeaderLabel are aligned to the start if it's placed in Gtk.Box.
However, the primary text is not aligned to the start if the label is placed in Gtk.Popover.

### Steps to Reproduce

1. Clone, compile, and run the following demo app: https://github.com/ryonakano/junk/tree/headerlabel-test
2. Click the "Open Popover" button to see the style differences between Granite.HeaderLabel in Gtk.Box and Granite.HeaderLabel in Gtk.Popover

### Expected Behavior

The primary and secondary text in Granite.HeaderLabel should be aligned to the start, regardless of the widget where it is placed.

### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
N/A
```

### Hardware Info

Not relevant

**Comment by ryonakano** [comment]
I thought we can simply fix this issue by removing this line which was introduced in #1233:

https://github.com/elementary/stylesheet/blob/be3742f3198541ff794c2c65469ea8dcf5f19367/src/gtk-4.0/widgets/_popovers.scss#L80

Then I can confirm the issue is fixed:

![image](https://github.com/user-attachments/assets/cab3fe99-5f30-4b4c-a88a-62edf34eb146)

However, I noticed this fix introduce regressions in other places like Display plug:

![Image](https://github.com/user-attachments/assets/eb73b356-c2a7-4008-ac6f-cc85df4cf20c)

**Comment by danirabbit** [comment]
Yeah it's trying to be overly clever tbh. We should remove it and then we'll have to fix the places where it breaks but it'll be more right

**Comment by ryonakano** [comment]
@danirabbit Possibly you fixed this in #1325?

**Comment by danirabbit** [comment]
@ryonakano thank you! I was looking for an issue report for this
